### PR TITLE
VZ-8843 Introduce the Thanos component to the Verrazzano install/upgrade

### PR DESCRIPTION
**Pull request by davisgomes**
- Introduce the Thanos Third party helm chart v12.1.1 (including common dependency)
- Create an install component for Thanos
- Add 0.28 thanos image to the bom
- Add a thanos field to the component section of the Verrazzano CR

### CODE REVIEW

**Review by desagar — APPROVED** [review]

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go [hunk 1]
```diff
@@ -1030,6 +1034,20 @@ type ArgoCDComponent struct {
...
 
+// ThanosComponent specifies the Thanos configuration.
+type ThanosComponent struct {
+	// If true, then Thanos will be installed.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+	// List of Overrides for the default `values.yaml` file for the component Helm chart. Overrides are merged together,
+	// but in the event of conflicting fields, the last override in the list takes precedence over any others. You can
+	// find all possible values
+	// [here]( {{% release_source_url path=platform-operator/thirdparty/charts/argo-cd/values.yaml %}} )
```
Comment (fredtibbitts) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:1045]: Copy/paste "argo-cd"

File: platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go [hunk 2]
```diff
@@ -948,6 +952,20 @@ type VeleroComponent struct {
...
 
+// ThanosComponent specifies the Thanos configuration.
+type ThanosComponent struct {
+	// If true, then Thanos will be installed.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+	// List of Overrides for the default `values.yaml` file for the component Helm chart. Overrides are merged together,
+	// but in the event of conflicting fields, the last override in the list takes precedence over any others. You can
+	// find all possible values
+	// [here]( {{% release_source_url path=platform-operator/thirdparty/charts/velero/values.yaml %}} )
```
Comment (fredtibbitts) [inline platform-operator/apis/verrazzano/v1beta1/verrazzano_types.go:963]: Copy/paste "velero"

File: platform-operator/verrazzano-bom.json [hunk 3]
```diff
@@ -777,6 +777,25 @@
           ]
         }
       ]
+    },
+    {
+      "name": "thanos",
+      "version": "0.28.0",
```
Comment (fredtibbitts) [inline platform-operator/verrazzano-bom.json:783]: Minor nit, looks like everywhere else the version here starts with "v" if the tag starts with "v", so it should be "v0.28.0".

File: pkg/bom/bom.go [hunk 4]
```diff
@@ -116,6 +116,11 @@ type BomImage struct {
 	// if the full image name is myreg.io/foo/bar/myimage:v1.0 the value of this key will be "myreg.io/foo/bar".
 	// See `image.repository` in the external-dns component
 	HelmRegistryAndRepoKey string `json:"helmRegistryAndRepoKey"`
+
+	// HelmRepoAndNameKey is a helm Key which stores the repo and name of the image path.  For example,
+	// if the full image name is myreg.io/foo/bar/myimage:v1.0 the value of this key will be "foo/bar/myimage".
+	// See `image.repository` in the thanos component
+	HelmRepoAndNameKey string `json:"helmRepoAndNameKey"`
```
Comment (fredtibbitts) [inline pkg/bom/bom.go:123]: I don't think you need to add a new key here. I think the existing `HelmFullImageKey` already does what you need.

Comment (davisgomes) [reply pkg/bom/bom.go:123]: Thanks. I missed this when setting up the bom

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**